src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -512,6 +513,175 @@ pass::EliminateConcat::EliminateConcat() {
     this->register_matcher(m, callback);
 }
 
+pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
+    MATCHER_SCOPE(EliminateConcatStridedSlice);
+    auto pattern_concat = pattern::wrap_type<ov::op::v0::Concat>(ov::pass::pattern::has_static_rank());
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_map();
+        const auto concat = ov::as_type_ptr<ov::op::v0::Concat>(pattern_map.at(pattern_concat));
+
+        const auto concat_axis =
+            ov::util::normalize(concat->get_axis(), concat->get_output_partial_shape(0).rank().get_length());
+
+        const auto concat_users = concat->get_users();
+        if (concat_users.size() == 1)
+            return false;
+        auto concat_inputs = concat->inputs();
+
+        std::vector<std::tuple<std::shared_ptr<Node>, int64_t, int64_t>> slice_out_index_in_concat;
```

Comment:
minor (optional):
probably it worth to introduce some alias for std::vector<std::tuple<std::shared_ptr<Node>, int64_t, int64_t>>
using ALIAS_NAME =  std::vector<std::tuple<std::shared_ptr<Node>, int64_t, int64_t>>